module/scheduler_components/simple_remaster_manager.cpp

```diff
@@ -0,0 +1,129 @@
+#include "module/scheduler_components/simple_remaster_manager.h"
+
+#include <glog/logging.h>
+
+namespace slog {
+
+SimpleRemasterManager::SimpleRemasterManager(
+    shared_ptr<Storage<Key, Record>> storage,
+    shared_ptr<TransactionMap> all_txns)
+  : storage_(storage), all_txns_(all_txns) {}
+
+VerifyMasterResult
+SimpleRemasterManager::VerifyMaster(const TxnReplicaId txn_replica_id) {
+  auto& txn_holder = all_txns_->at(txn_replica_id);
+  auto& keys = txn_holder.KeysInPartition();
+  if (keys.empty()) {
+    return VerifyMasterResult::VALID;
+  }
+
+  auto txn = txn_holder.GetTransaction();
+  auto& txn_master_metadata = txn->internal().master_metadata();
+  if (txn_master_metadata.empty()) { // This should only be the case for testing
+    LOG(WARNING) << "Master metadata empty: txn id " << txn->internal().id();
+    return VerifyMasterResult::VALID;
+  }
+
+  // Determine which local log this txn is from. Since only single home or
+  // lock only txns, all keys will have same master
+  auto local_log_machine_id = txn_master_metadata.begin()->second.master();
+
+  // Block this txn behind other txns from same local log
+  // TODO: check the counters now? would abort earlier
+  if (blocked_queue_.count(local_log_machine_id)) {
+    blocked_queue_.at(local_log_machine_id).push_back(txn_replica_id);
```

Comment:
nit: as you already checked with `count()`, use `[]` instead of `at` to avoid calling `count()` (inside `at()`) again